src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb

```diff
@@ -705,5 +707,28 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
         End Class
 #End Region
 
+        Public Class SerializationProvider
+
+            Public Shared Sub Serialize(stream As Stream, value As Object)
+                If value Is Nothing Then
+                    Return
+                End If
+                Using writer As New BinaryWriter(stream, Encoding.UTF8, True)
+                    Dim valueType = value.GetType()
+                    writer.Write(valueType.AssemblyQualifiedName)
+                    writer.Flush()
+                    Call New DataContractSerializer(valueType).WriteObject(stream, value)
+                End Using
```

Comment:
I would invert these lines personally. I think it's clearer:

```suggestion
                End Using
                Call New DataContractSerializer(valueType).WriteObject(stream, value)
```

Reply:
The reason this is done is because deserialization initializes `valueType` in the using statement. Otherwise, you would need a sentinel value (such as Nothing) and then assign it in the using. Both serialization and deserialization were written in a way to appear similar to each other. So, I opted for that consistency over making the usings completely pure. Either is straightforward enough to read. It is 3 or 4 lines of code. Pulling `valueType` out of the using means it is declared earlier than it is used, which is my preference to not do.